spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# See testable_service/integration_test.py and spinnaker_testing/spinnaker.py
+# for more details.
+#
+# The Http(s) test will be used to (more) thoroughly test the GCP L7 load
+# balancer Upsert and Delete commands, and will eventually be expanded to
+# include server group tests. This test operates on a L7 Lb with zero server
+# groups attached.
+
+# Standard python modules.
+from OpenSSL import crypto
+import copy
+import json
+import sys
+import time
+import uuid
+
+# citest modules.
+from citest.base import ExecutionContext
+import citest
+import citest.gcp_testing as gcp
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+HTTP_PROXY_PREFIX = 'target-http-proxy'
+HTTPS_PROXY_PREFIX = 'target-https-proxy'
+SCOPES = [gcp.COMPUTE_READ_WRITE_SCOPE]
+GCE_URL_PREFIX = 'https://www.googleapis.com/compute/v1/projects/'
+
+class GoogleHttpLoadBalancerTestScenario(sk.SpinnakerTestScenario):
+  '''Defines the tests for L7 Load Balancers.
+  '''
+
+  MINIMUM_PROJECT_QUOTA = {
+    'INSTANCE_TEMPLATES': 1,
+    'BACKEND_SERVICES': 3,
+    'URL_MAPS': 1,
+    'HEALTH_CHECKS': 1,
+    'IN_USE_ADDRESSES': 2,
+    'SSL_CERTIFICATES': 2,
+    'TARGET_HTTP_PROXIES': 1,
+    'TARGET_HTTPS_PROXIES': 1,
+    'FORWARDING_RULES': 2
+  }
+
+  MINIMUM_REGION_QUOTA = {
+      'CPUS': 2,
+      'IN_USE_ADDRESSES': 2,
+      'INSTANCE_GROUP_MANAGERS': 1,
+      'INSTANCES': 2,
+  }
+
+  @classmethod
+  def new_agent(cls, bindings):
+    '''Implements citest.service_testing.AgentTestScenario.new_agent.'''
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 600
+    return agent
+
+
+  @classmethod
+  def initArgumentParser(cls, parser, defaults=None):
+    '''Initialize command line argument parser.
+
+    Args:
+      parser: argparse.ArgumentParser
+    '''
+    super(GoogleHttpLoadBalancerTestScenario, cls).initArgumentParser(
+      parser, defaults=defaults)
+
+
+  def __init__(self, bindings, agent=None):
+    '''Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    '''
+    super(GoogleHttpLoadBalancerTestScenario, self).__init__(bindings, agent)
+
+    bindings = self.bindings
+
+    self.__lb_detail = 'lb'
+    self.TEST_APP = bindings['TEST_APP']
+    self.__lb_name = '{app}-{stack}-{detail}'.format(
+        app=bindings['TEST_APP'], stack=bindings['TEST_STACK'],
+        detail=self.__lb_detail)
+    self.__proto_hc = {
+      'name': 'basic-http-check',
+      'requestPath': '/',
+      'port': 80,
+      'checkIntervalSec': 1,
+      'timeoutSec': 1,
```

Comment:
I like to make these values different so if the code transposes them you'll catch it.